### PR TITLE
replace logrus with meshkit/logger in mesheryctl

### LINKED ISSUE #4705: [mesheryctl] Replace logrus with meshkit logger
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

mesheryctl is using https://github.com/sirupsen/logrus everywhere

#### Desired Behavior

`Logrus` needs to be replaced with `meshkit` logger

---

#### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by ayanchoudhary**
**Description**
This PR replaces the logrus package with meshkit/logger in mesheryctl

This PR provides partial resolution of #4705

**Notes for Reviewers**
This PR is the first part towards completing issue #4705. The logrus logger has been replaced with suitable alternative from the meshkit/logger package and any incompatible methods/levels have been left unchanged.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks, @ayanchoudhary. Good stuff. Given that this is only a partial fix, will you update the verbiage used in the PR description from:

```
This PR fixes #4705
```

to 

```
This PR provides partial resolution of #4705
```

The reason for this update is because GitHub will automatically close the #4705 issue when this PR merges unless the verbiage is changed.

**Comment by piyushsingariya** [comment]
Merge conflict alert!

File: mesheryctl/pkg/utils/formatter.go [hunk 1]
```diff
@@ -1,6 +1,10 @@
 package utils
 
 import (
+	"os"
+
+	"github.com/layer5io/meshkit/logger"
+	"github.com/sirupsen/logrus"
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/formatter.go:7]: multiple imports of logrus, please remove one instance.

File: mesheryctl/pkg/utils/formatter.go [hunk 2]
```diff
@@ -18,3 +22,15 @@ func SetupLogrusFormatter() {
...
 	log.SetFormatter(new(TerminalFormatter))
 }
+
+// Initialize Meshkit Logger instance
+func MeshkitLogger() (logger.Handler, error) {
+	log, err := logger.New("mesheryctl", logger.Options{
+		Format: logger.SyslogLogFormat,
+	})
+	if err != nil {
+		logrus.Error(err)
```
Comment (hexxdump) [inline mesheryctl/pkg/utils/formatter.go:32]: We should get away from this logrus instance as well, right?

File: mesheryctl/internal/cli/root/app/onboard.go [hunk 3]
```diff
@@ -41,6 +41,9 @@ var onboardCmd = &cobra.Command{
 		var err error
 		client := &http.Client{}
 
+		// get logger instance
+		log, _ := utils.MeshkitLogger()
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/app/onboard.go:45]: can we initialize logger instance in root.go, instead of each sub command ?

Comment (ayanchoudhary) [reply mesheryctl/internal/cli/root/app/onboard.go:45]: I was thinking of the same initially but having a single instance will lead to passing that instance through many functions across different levels and will introduce unnecessary complexity in the flow, hence I went with initializing different instances something along the lines of what logrus provides

File: mesheryctl/pkg/utils/formatter.go (hunk 2 above)
Comment (piyushsingariya) [reply mesheryctl/pkg/utils/formatter.go]: Logrus needs to be completely wiped out, so best to replace these too.

**Review by hexxdump — APPROVED** [review]
LGTM

**Comment by piyushsingariya** [comment]
Tests failing! @ayanchoudhary

**Comment by leecalcote** [comment]
Re-running build checks...

**Comment by navendu-pottekkat** [comment]
@ayanchoudhary Could you check the failing unit tests? Since this is a significant change in the logs, the assertions are bound to throw some errors.

**Comment by ayanchoudhary** [comment]
Sure, I will have a look at those

**Comment by piyushsingariya** [comment]
> Sure, I will have a look at those

reminder!

**Comment by leecalcote** [comment]
@ayanchoudhary will you be incorporating the feedback above?

**Comment by ayanchoudhary** [comment]
I am kinda pre occupied with something at the moment, so it may be a while before I incorporate the above feedback

**Comment by leecalcote** [comment]
@ayanchoudhary thanks for this quick note. I complete understand. We'll carry this forward to closure in the meantime.

@navendu-pottekkat @piyushsingariya @hexxdump, will one of you pick up the slack here?

**Comment by piyushsingariya** [comment]
@leecalcote I'll need [this](https://github.com/layer5io/meshkit/pull/140) to merge to manage unit tests here in this PR.

**Comment by leecalcote** [comment]
> @leecalcote I'll need [this PR](https://github.com/layer5io/meshkit/pull/140) to merge to handle unit tests here.

Sounds good, @piyushsingariya. Merged. 👍

**Comment by piyushsingariya** [comment]
@leecalcote a new release is required of meshkit and dependencies will also needed to be updated.

**Comment by leecalcote** [comment]
@piyushsingariya, MeshKit v0.5.1 includes the terminal formatter support.

**Comment by piyushsingariya** [comment]
> @piyushsingariya, MeshKit v0.5.1 includes the terminal formatter support.

Ohh yeah, I was wrapping up this PR yesterday, and found few mistakes I made in meshkit, a new PR has been opened up with fixes, after that I can complete the work here.

**Comment by leecalcote** [comment]
@piyushsingariya we should be in a position to move this PR forward now.

**Comment by piyushsingariya** [comment]
Closing this PR, opening a new one!

**Pull request closed without merging**